platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/sync/SingleNodeSyncGossip.java

```diff
@@ -0,0 +1,287 @@
+
+        this.eventIntakeLambda = Objects.requireNonNull(eventIntakeLambda);
+
+        syncConfig = platformContext.getConfiguration().getConfigData(SyncConfig.class);
+
+        final ParallelExecutor shadowgraphExecutor = PlatformConstructor.parallelExecutor(threadManager);
+        thingsToStart.add(shadowgraphExecutor);
+        syncShadowgraphSynchronizer = new ShadowGraphSynchronizer(
+                shadowGraph,
+                addressBook.getSize(),
+                syncMetrics,
+                consensusRef::get,
+                eventTaskCreator::syncDone,
+                eventTaskCreator::addEvent,
+                syncManager,
+                shadowgraphExecutor,
+                // don't send or receive init bytes if running sync as a protocol. the negotiator handles this
+                !syncConfig.syncAsProtocolEnabled(),
+                () -> {});
+
+        clearAllPipelines = new LoggingClearables(
+                RECONNECT.getMarker(),
+                List.of(
+                        Pair.of(intakeQueue, "intakeQueue"),
+                        Pair.of(eventMapper, "eventMapper"),
+                        Pair.of(shadowGraph, "shadowGraph")));
+
+        final BasicConfig basicConfig = platformContext.getConfiguration().getConfigData(BasicConfig.class);
+
+        final Duration hangingThreadDuration = basicConfig.hangingThreadDuration();
+
+        syncProtocolThreads.add(new StoppableThreadConfiguration<>(threadManager)
+                .setPriority(Thread.NORM_PRIORITY)
+                .setNodeId(selfId.id())
+                .setComponent(PLATFORM_THREAD_POOL_NAME)
+                .setOtherNodeId(selfId.id())
+                .setThreadName("SingleNodeNetworkSync")
+                .setHangingThreadPeriod(hangingThreadDuration)
+                .setWork(new SingleNodeNetworkSync(
+                        updatePlatformStatus, eventTaskCreator::createEvent, () -> 0, selfId.id()))
+                .build());
+
+        thingsToStart.add(() -> syncProtocolThreads.forEach(StoppableThread::start));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected boolean unidirectionalConnectionsEnabled() {
+        return false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void stop() {
+        super.stop();
+        // wait for all existing syncs to stop. no new ones will be started, since gossip has been halted, and
+        // we've fallen behind
+        for (final StoppableThread thread : syncProtocolThreads) {
+            thread.stop();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected CriticalQuorum buildCriticalQuorum() {
+        return new CriticalQuorumImpl(platformContext.getMetrics(), selfId.id(), addressBook);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected FallenBehindManagerImpl buildFallenBehindManager() {
+        return new FallenBehindManagerImpl(
```

Comment:
fixed

Reply:
Actually, it is necessary, adding it back in. Allowing this to be null would force me to add null checks in other parts of the code, and I think it's more elegant not to have to do that... especially since the 1 node network is not the primary use case.

If we end up having a different gossip implementation for a single node network in the long term, we can certainly do cleanup. E.g. perhaps the single node implementation wouldn't bother extending the abstract class. But for now it's simpler just to construct an instance and return it.